### PR TITLE
Added C++ client to be compiled with other targets

### PR DESCRIPTION
**Pull request by Blyron**
Add support to compile C++ Example using default cmake process

- - -
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7893)

### CODE REVIEW

File: Examples/CMakeLists.txt [hunk 1]
```diff
@@ -1 +1,73 @@
-add_subdirectory (CppClient)
+set (
+  CARLA_EXAMPLE_CPP_CLIENT_SOURCE_PATH
+  ${CARLA_WORKSPACE_PATH}/Examples/CppClient
+)
+
+set (
```
Comment (MarcelPiNacy-CVC) [inline Examples/CMakeLists.txt:6]: Do we know if only passing carla-client as a dependency is enough? In theory carla-client should already be including all of these.

File: Examples/CMakeLists.txt [hunk 2]
```diff
@@ -1 +1,73 @@
...
+  Boost::asio
+  Boost::python
+  Boost::geometry
+  Boost::gil
+  RecastNavigation::Recast
+  RecastNavigation::Detour
+  RecastNavigation::DetourCrowd
+)
+
+if (BUILD_EXAMPLES)
```
Comment (MarcelPiNacy-CVC) [inline Examples/CMakeLists.txt:21]: This if can be removed: we check for BUILD_EXAMPLES being on in the top CMakeLists.txt file, in which case all subprojects in Examples/ will be skipped.

File: Examples/CMakeLists.txt [hunk 3]
```diff
@@ -1 +1,73 @@
...
+    ${CARLA_EXAMPLE_CPP_CLIENT_SOURCE_PATH}/*.h
+  )
+  
+  carla_add_executable (
+    carla-example-client
+    "Build the CARLA C++ client example."
+    ${CARLA_EXAMPLE_CPP_SOURCES}
+  )
+    
+  add_dependencies (
```
Comment (MarcelPiNacy-CVC) [inline Examples/CMakeLists.txt:52]: This add_dependencies is not needed, as target_link_libraries already implicitly adds the dependency.

File: Examples/CMakeLists.txt [hunk 4]
```diff
@@ -1 +1,73 @@
...
+    LANGUAGES
+      CXX
+    VERSION
+      ${CARLA_VERSION}
+  )
+
+  file (
+    GLOB_RECURSE 
+    CARLA_EXAMPLE_CPP_SOURCES
+    CONFIGURE_DEPENDS
```
Comment (MarcelPiNacy-CVC) [inline Examples/CMakeLists.txt:35]: Is CONFIGURE_DEPENDS needed here?

File: Examples/CMakeLists.txt [hunk 5]
```diff
@@ -1 +1,73 @@
...
+  )
+
+  file (
+    GLOB_RECURSE 
+    CARLA_EXAMPLE_CPP_SOURCES
+    CONFIGURE_DEPENDS 
+    ${CARLA_EXAMPLE_CPP_CLIENT_SOURCE_PATH}/*.cpp
+  )
+
+  file (
```
Comment (MarcelPiNacy-CVC) [inline Examples/CMakeLists.txt:39]: CARLA_EXAMPLE_CPP_HEADERS is not referenced in carla_add_executable.

File: Examples/CMakeLists.txt (hunk 4 above)
Comment (MarcelPiNacy-CVC) [reply Examples/CMakeLists.txt:35]: Nevermind, this is indeed correct. We should update other source-related GLOB calls in CARLA.

File: Examples/CMakeLists.txt (hunk 2 above)
Comment (Blyron) [reply Examples/CMakeLists.txt]: updated

File: Examples/CMakeLists.txt (hunk 3 above)
Comment (Blyron) [reply Examples/CMakeLists.txt]: Updated

File: Examples/CMakeLists.txt (hunk 4 above)
Comment (Blyron) [reply Examples/CMakeLists.txt]: Okay

File: Examples/CMakeLists.txt (hunk 5 above)
Comment (Blyron) [reply Examples/CMakeLists.txt]: Updated

**Review by MarcelPiNacy-CVC — APPROVED** [review]

**Pull request merged**